core/src/main/kotlin/com/malinskiy/marathon/execution/device/DeviceActor.kt

```diff
@@ -187,21 +211,19 @@ class DeviceActor(private val devicePoolId: DevicePoolId,
         job = async {
             try {
                 device.execute(configuration, devicePoolId, batch, result, progressReporter)
-            } catch (e: DeviceLostException) {
-                logger.error(e) { "Critical error during execution" }
+            } catch (exc: TestBatchTimeoutException) {
```

Comment:
if possible, please implement this sort of device-management behavior in android-specific class, like I did for iOS. unless you're sure that ios and android branches will never merge.

Reply:
Hmm, I see your point, but I'm not sure how to implement it. As for merging with iOS - not sure, I think the best option is to merge with the main project's develop branch

Reply:
Let's discuss how to do it in the best way.

Reply:
The timeout exception will not be happening anymore in develop: we're getting rid of ddmlib and blocking code. This means that the timeout will be handled by coroutines, not some for-loop that throws TimeoutException. I can help you integrate with the new changes if you wish to merge with develop

Reply:
Yes, I think that is the next logical step - to use develop branch for Android, then maybe for iOS.